src/Compilers/CSharp/Portable/Lowering/StateMachineRewriter/StateMachineRewriter.cs

```diff
@@ -316,12 +316,9 @@ protected BoundStatement GenerateParameterStorage(LocalSymbol stateMachineVariab
                 }
             }
 
-            bodyBuilder.Add(GenerateStateMachineCreation(stateMachineVariable, frameType));
             var builtBody = bodyBuilder.ToImmutableAndFree();
             ImmutableArray<BoundStatement> newBody = LocalRewriter.ConstructNullCheckedStatementList(method.Parameters, builtBody, F);
```

Comment:
I know you didn't change this in this PR, but it would be good to rename `ConstructNullCheckedStatementList` to `TryConstructNullCheckedStatementList`, or something similar that conveys that the body it returns could be `default`, as it's not a nullable type.

Reply:
Renamed the method to better fit. Thanks.